client/src/components/Map/UCIMap.js

```diff
@@ -232,28 +246,41 @@ export default class UCIMap extends PureComponent {
 
         // Tracks courses that have already been pinned on the map, so there are no duplicates
         let pinnedCourses = new Set();
-        let index = 0;
+        let pins = {};
+        let courses = new Set();
 
         // Filter out those in a different schedule or those not on a certain day (mon, tue, etc)
         this.state.eventsInCalendar
             .filter(
                 (event) =>
                     !(
-                        event.isCustomEvent ||
-                        !event.scheduleIndices.includes(AppStore.getCurrentScheduleIndex()) ||
-                        !event.start.toString().includes(DAYS[this.state.day])
+                        (
+                            event.isCustomEvent ||
+                            !event.scheduleIndices.includes(AppStore.getCurrentScheduleIndex()) ||
+                            !event.start.toString().includes(DAYS[this.state.day]) ||
+                            courses.has(event.sectionCode) || // Remove duplicate courses that appear in the calendar
+                            !courses.add(event.sectionCode)
```

Comment:
Won't this always return true, since the set isn't empty?

Reply:
it's a trick and I agree it's not clear but I couldn't think of an easier way to add filtering of the sectioncode without going through the array again.
if !event.scheduleIndices.includes(AppStore.getCurrentScheduleIndex()) or !event.start.toString().includes(DAYS[this.state.day]) conditions are true then it is negated and returns false to the filter (if the schedule indices does not include the current index or the event does not start in the current day). If those are passed, check if the courses set if the event has already been added, if true it return false after negation and filter skips it else it adds to the set and return false, which actually returns true.

a better way to handle this is to "Use addedCourses instead of eventsInCalendar to create markers"  as #112 mentioned